### PR TITLE
Add CybelAngelEventCollector

### PR DESCRIPTION
**Pull request by GuyAfik**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CybelAngel/Integrations/CybleAngelEventCollector/CybleAngelEventCollector.py [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+        params = {
+            "start-date": start_date,
+            "end-date": end_date
+        }
+        reports = self.http_request(method='GET', url_suffix="/api/v2/reports", params=params).get("reports") or []
+        return reports[:limit]
+
+    def get_access_token(self) -> str:
+        """
+       Obtains access and refresh token from server.
```
Comment (GuyAfik) [inline Packs/CybelAngel/Integrations/CybleAngelEventCollector/CybleAngelEventCollector.py:60]:
```suggestion
       Obtains access and refresh token from CybleAngel server.
```

File: Packs/CybelAngel/Integrations/CybleAngelEventCollector/CybleAngelEventCollector.py [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+        }
+        demisto.debug(f'updating access token')
+        set_integration_context(context=integration_context)
+
+        return access_token
+
+
+def is_token_expired(token_initiate_time: float, token_expiration_seconds: float) -> bool:
+    """
+    Check whether a token has expired. a token considered expired if it has been reached to its expiration date in
```
Comment (GuyAfik) [inline Packs/CybelAngel/Integrations/CybleAngelEventCollector/CybleAngelEventCollector.py:112]:
```suggestion
    Checks whether a token has expired. a token considered expired if it has been reached to its expiration date in
```

File: Packs/CybelAngel/Integrations/CybelAngelEventCollector/CybelAngelEventCollector_description.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+CybelAngel is a cybersecurity firm specializing in external attack surface protection and management.
```
Comment (JudahSchwartz) [inline Packs/CybelAngel/Integrations/CybelAngelEventCollector/CybelAngelEventCollector_description.md:1]: shouldnt this description be the same as is in the readme? There should be details about what is being collected in this collector

File: Packs/CybelAngel/Integrations/CybelAngelEventCollector/README.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+## Commands
+
+You can execute these commands from the Cortex XSOAR CLI, as part of an automation, or in a playbook.
+After you successfully execute a command, a DBot message appears in the War Room with the command details.
+
+### cybelangel-get-events
+
+***
+Collect reports from CybelAngel. Used mainly for debugging.
```
Comment (JudahSchwartz) [inline Packs/CybelAngel/Integrations/CybelAngelEventCollector/README.md:29]: Does this call sendevent to xsiam? If not it should be clear the event isnt "collected"

File: Packs/CybelAngel/Integrations/CybelAngelEventCollector/CybelAngelEventCollector.py [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+    last_run.update(
+        {
+            LastRun.LATEST_REPORT_TIME: latest_report_time,
+            LastRun.LATEST_FETCHED_REPORTS_IDS: latest_fetched_report_ids
+        }
+    )
+    return reports, last_run
+
+
+def get_events_command(client: Client, args: dict[str, Any]) -> CommandResults:
```
Comment (JudahSchwartz) [inline Packs/CybelAngel/Integrations/CybelAngelEventCollector/CybelAngelEventCollector.py:197]: docstring

File: Packs/CybelAngel/Integrations/CybelAngelEventCollector/CybelAngelEventCollector.py [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+    Tests that the authentication to the api is ok.
+    """
+    client.get_reports(
+        start_date=(datetime.now() - timedelta(days=1)).strftime(DATE_FORMAT),
+        end_date=datetime.now().strftime(DATE_FORMAT)
+    )
+    return "ok"
+
+
+def fetch_events(client: Client, first_fetch: str, last_run: dict, max_fetch: int) -> tuple[List[dict[str, Any]], dict[str, Any]]:
```
Comment (JudahSchwartz) [inline Packs/CybelAngel/Integrations/CybelAngelEventCollector/CybelAngelEventCollector.py:166]: docstring

File: Packs/CybelAngel/Integrations/CybelAngelEventCollector/CybelAngelEventCollector.py [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+    last_run_time = last_run.get(LastRun.LATEST_REPORT_TIME)
+    if not last_run_time:
+        last_run_time = dateparser.parse(first_fetch).strftime(DATE_FORMAT)  # type: ignore[union-attr]
+        if not last_run_time:
+            raise ValueError('First fetch is not valid')
```
Comment (JudahSchwartz) [inline Packs/CybelAngel/Integrations/CybelAngelEventCollector/CybelAngelEventCollector.py:171]: add a log

**Review by JudahSchwartz — APPROVED** [review]
masterpiece 👨‍🎨

**Comment by JudahSchwartz** [comment]
tag doc team please

File: Packs/CybelAngel/Integrations/CybelAngelEventCollector/CybelAngelEventCollector_description.md (hunk 3 above)
Comment (tschanfeld) [reply Packs/CybelAngel/Integrations/CybelAngelEventCollector/CybelAngelEventCollector_description.md:1]: change it to 'CybelAngel Event Collector collects reports from the CybelAngel platform, which specializes in external attack surface protection and management.' ?? or change both to 'CybelAngel Event Collector receives reports from the CybelAngel platform, which specializes in external attack surface protection and management.' ?

Comment (GuyAfik) [reply Packs/CybelAngel/Integrations/CybelAngelEventCollector/CybelAngelEventCollector_description.md:1]: @tschanfeld The second one

File: Packs/CybelAngel/Integrations/CybelAngelEventCollector/README.md (hunk 4 above)
Comment (GuyAfik) [reply Packs/CybelAngel/Integrations/CybelAngelEventCollector/README.md:29]: @tschanfeld yes
@JudahSchwartz we'll document it

File: Packs/CybelAngel/Integrations/CybelAngelEventCollector/CybelAngelEventCollector.py [hunk 8]
```diff
@@ -0,0 +1,289 @@
...
+            LastRun.LATEST_REPORT_TIME: latest_report_time,
+            LastRun.LATEST_FETCHED_REPORTS_IDS: latest_fetched_report_ids
+        }
+    )
+    return reports, last_run
+
+
+def get_events_command(client: Client, args: dict[str, Any]) -> CommandResults:
+    """
+    Get events from Cybel Angel, used mainly for debugging purposes
```
Comment (tschanfeld) [inline Packs/CybelAngel/Integrations/CybelAngelEventCollector/CybelAngelEventCollector.py:225]: @GuyAfik do you want to change this to 'Send events from CybelAngel to XSIAM. Used mainly for debugging.'

Comment (GuyAfik) [reply Packs/CybelAngel/Integrations/CybelAngelEventCollector/CybelAngelEventCollector.py:225]: i will update it yes, thanks

**Pull request merged**